packages/moonstone/TooltipDecorator/TooltipDecorator.js

```diff
@@ -331,7 +331,9 @@ const TooltipDecorator = hoc(defaultConfig, (config, Wrapped) => {
 				this.clientRef = null;
 				currentTooltip = null;
 				this.showTooltipJob.stop();
-				this.setState({showing: false});
+				if (this.state.showing) {
+					this.setState({showing: false});
```

Comment:
Should we use functional `setState()` here and in `showTooltipJob()` above? Could generalize them with something like:

```
const toggleShowing = (showing) => (state) => state.showing !== showing ? {showing} : null;
// ...
this.setState(toggleShowing(false));
```

Reply:
I like the readability of that and the idea of functional `setState`, but I've been avoiding it since it creates (or adds time to) a react tree reconciliation update simply when utilizing it - even when nothing should actually update.

Reply:
I think we need resolve this issue (using function `setState`) in the longer term but the usage here is trivial so let's go ahead with the current impl.